src/main/java/seedu/exercise/logic/commands/ScheduleRegimeCommand.java

```diff
@@ -49,16 +55,18 @@ private void checkExistenceOfRegime(Model model) throws CommandException {
 
     /**
      * Checks for scheduling conflicts and returns a valid schedule if no conflicts are found.
-     *
-     * @throws ScheduleException when there is another regime scheduled on the same date as {@code dateToSchedule}
+     * If a conflict is found, returns a null schedule
      */
-    private Schedule checkSchedulingConflict(Model model) throws ScheduleException {
+    private Schedule checkSchedulingConflict(Model model) {
```

Comment:
It feels weird that after the check returns a Schedule. Perhaps you can separate out the initialization of schedule and the checking of conflicts into separate methods

Reply:
changed method name to reflect what the method does better